packages/read-model/src/Product/Detail/ProductDetailViewElasticsearchFactory.php

```diff
@@ -0,0 +1,167 @@
+     */
+    protected $parameterViewFactory;
+
+    /**
+     * @var \Shopsys\ReadModelBundle\Brand\BrandViewFactory
+     */
+    protected $brandViewFactory;
+
+    /**
+     * @var \Shopsys\FrameworkBundle\Model\Product\ProductOnCurrentDomainElasticFacade
+     */
+    protected $productOnCurrentDomainElasticFacade;
+
+    /**
+     * @var \Shopsys\ReadModelBundle\Product\Listed\ListedProductViewElasticFacade
+     */
+    protected $listedProductViewElasticFacade;
+
+    /**
+     * @param \Shopsys\ReadModelBundle\Image\ImageViewFacadeInterface $imageViewFacade
+     * @param \Shopsys\FrameworkBundle\Model\Customer\User\CurrentCustomerUser $currentCustomerUser
+     * @param \Shopsys\ReadModelBundle\Product\Action\ProductActionViewFactory $productActionViewFactory
+     * @param \Shopsys\ReadModelBundle\Parameter\ParameterViewFactory $parameterViewFactory
+     * @param \Shopsys\ReadModelBundle\Brand\BrandViewFactory $brandViewFactory
+     * @param \Shopsys\ReadModelBundle\Product\Listed\ListedProductViewElasticFacade $listedProductViewElasticFacade
+     * @param \Shopsys\FrameworkBundle\Model\Product\ProductOnCurrentDomainElasticFacade $productOnCurrentDomainElasticFacade
+     */
+    public function __construct(
+        ImageViewFacadeInterface $imageViewFacade,
+        CurrentCustomerUser $currentCustomerUser,
+        ProductActionViewFactory $productActionViewFactory,
+        ParameterViewFactory $parameterViewFactory,
+        BrandViewFactory $brandViewFactory,
+        ListedProductViewElasticFacade $listedProductViewElasticFacade,
+        ProductOnCurrentDomainElasticFacade $productOnCurrentDomainElasticFacade
+    ) {
+        $this->imageViewFacade = $imageViewFacade;
+        $this->currentCustomerUser = $currentCustomerUser;
+        $this->productActionViewFactory = $productActionViewFactory;
+        $this->parameterViewFactory = $parameterViewFactory;
+        $this->brandViewFactory = $brandViewFactory;
+        $this->productOnCurrentDomainElasticFacade = $productOnCurrentDomainElasticFacade;
+        $this->listedProductViewElasticFacade = $listedProductViewElasticFacade;
+    }
+
+    /**
+     * @param array $productArray
+     * @return \Shopsys\ReadModelBundle\Product\Detail\ProductDetailView
+     */
+    public function createFromProductArray(array $productArray): ProductDetailView
+    {
+        $parameterViews = [];
+        foreach ($productArray['parameters'] as $parameterArray) {
+            $parameterViews[] = $this->parameterViewFactory->createFromParameterArray($parameterArray);
+        }
+
+        $accessories = $this->listedProductViewElasticFacade->createFromArray(
+            $this->productOnCurrentDomainElasticFacade->getVisibleProductArrayByIds($productArray['accessories'])
+        );
+        $variants = $this->listedProductViewElasticFacade->createFromArray(
+            $this->productOnCurrentDomainElasticFacade->getVisibleProductArrayByIds($productArray['variants'])
+        );
```

Comment:
Then it would be nice to add `ListedProductViewElasticFacade::getAllVariants` so it is same as in `ListedProductViewFacade`

Reply:
Negative as above.